plugin/src/main/java/com/playmonumenta/scriptedquests/trades/TradeWindowOpenEvent.java

```diff
@@ -25,12 +25,14 @@ public class TradeWindowOpenEvent extends Event implements Cancellable {
 	private boolean mCancelled;
 
 	public static class Trade {
-		private MerchantRecipe mRecipe;
-		private @Nullable QuestActions mActions;
+		private final MerchantRecipe mRecipe;
+		private final @Nullable QuestActions mActions;
+		private final int mCount;
 
-		public Trade(MerchantRecipe recipe, @Nullable QuestActions actions) {
+		public Trade(MerchantRecipe recipe, @Nullable QuestActions actions, int count) {
```

Comment:
This will break the main plugin BTW. So you either should add back the old constructor, or update the main plugin to use this new constructor.

Updating the main plugin is probably the better option in case a trade with a count is ever changed by it. An idea would also be to create a kind of copy constructor so further additions in the future won't break the API again (e.g. `Trade(Trade original, MerchantRecipe newRecipe)`), or maybe add setters for all fields and only provide a normal copy constructor.

Also, the original result should be part of the trade event so that it too can be properly changed by the main plugin - or alternatively (and probably simpler), move the code that creates the fake trades to after the trade event so that it picks up any changes automatically.

Reply:
I'd advise making `public Trade(MerchantRecipe recipe, @Nullable QuestActions actions)` call `public Trade(MerchantRecipe recipe, @Nullable QuestActions actions, int count)` with a default count. This will avoid breaking any 3rd party plugins depending on SQ.

Reply:
Those plugins (if any) will just need to be updated when they update SQ - not only did the API change, but the behaviour too (assuming you have at least one trade where you use the count)

Reply:
Isn't the behavior unchanged for existing trades with the default count?